ckanext/dia_theme/less/icons.less

```diff
@@ -145,6 +145,12 @@
   .ckan-icon-background-position(9, "formatMedium");
 }
 
+//Fix to display geojson icon
+.format-label[data-format=geojson],
+.format-label[data-format*=geojson] {
+  background-position: -128px -62px;
```

Comment:
The stylesheet from the geo extension appears to override things, these seemed to be the min change to achieve the result.